src/zabapgit_transport.prog.abap

```diff
@@ -203,5 +183,140 @@ CLASS lcl_transport IMPLEMENTATION.
+
+    create_new_branch(
+      io_repository  = io_repository
+      iv_branch_name = lv_branch_name ).
+
+    CREATE OBJECT lo_stage
+      EXPORTING
+        iv_branch_name = lv_branch_name
+        iv_branch_sha1 = io_repository->get_sha1_remote( ).
+
+    ls_stage_objects = lcl_stage_logic=>get( io_repository ).
+
+    lt_object_statuses = io_repository->status( ).
+
+    stage_transport_objects(
+       it_transport_objects = it_transport_objects
+       io_stage             = lo_stage
+       is_stage_objects     = ls_stage_objects
+       it_object_statuses   = lt_object_statuses ).
+
+    ls_comment = generate_commit_message( is_transport_to_branch ).
+
+    io_repository->push( is_comment = ls_comment
+                         io_stage   = lo_stage ).
+  ENDMETHOD.
+
+  METHOD create_new_branch.
+    ASSERT iv_branch_name CP 'refs/heads/+*'.
+    TRY.
+        lcl_git_porcelain=>create_branch(
+          io_repo = io_repository
+          iv_name = iv_branch_name
+          iv_from = io_repository->get_sha1_local( ) ).
+
+        io_repository->set_branch_name( iv_branch_name ).
+      CATCH lcx_exception.
+        lcx_exception=>raise( 'Error when creating new branch').
+    ENDTRY.
+  ENDMETHOD.
+
+  METHOD generate_commit_message.
+    rs_comment-committer-name  = sy-uname.
+    rs_comment-committer-email = |{ rs_comment-committer-name }@localhost|.
+    rs_comment-comment         = is_transport_to_branch-commit_text.
+  ENDMETHOD.
+
+
+  METHOD stage_transport_objects.
+
+    DATA ls_transport_object TYPE tadir.
+    DATA ls_local_file TYPE ty_file_item.
+    DATA ls_object_status TYPE ty_result.
+
+    LOOP AT it_transport_objects INTO ls_transport_object.
+      READ TABLE it_object_statuses INTO ls_object_status
+        WITH KEY obj_name = ls_transport_object-obj_name
+                 obj_type = ls_transport_object-object.
+      IF sy-subrc <> 0.
+        lcx_exception=>raise( |Object { ls_transport_object-obj_name } not found in the local repository files | ).
+      ENDIF.
+
+      CASE ls_object_status-lstate.
+        WHEN gc_state-added OR gc_state-modified.
+          ASSERT ls_transport_object-delflag = abap_false.
+
+          READ TABLE is_stage_objects-local
+                INTO ls_local_file
+            WITH KEY item-obj_name = ls_transport_object-obj_name
+                     item-obj_type = ls_transport_object-object.
+          IF sy-subrc <> 0.
+            lcx_exception=>raise( |Object { ls_transport_object-obj_name } not found in the local repository files | ).
+          ENDIF.
+
+          io_stage->add(
+            iv_path     = ls_local_file-file-path
+            iv_filename = ls_local_file-file-filename
+            iv_data     = ls_local_file-file-data ).
+        WHEN gc_state-deleted.
+          ASSERT ls_transport_object-delflag = abap_true.
+          io_stage->rm(
+            iv_path     = ls_object_status-path
+            iv_filename = ls_object_status-filename ).
```

Comment:
No. It should not appear in bare status. This is the status which means that underlying object contain multiple statuses (e.g. some FM has several files: one include changed, another added - so the agregated status of FM is mixed). It can appear only in aggregations in the main page (see `_reduce_state` macro) and cannot be in `status()` results as it contain one file per line. I hope it clarifies ... So I guess just put `when others` - in case of exception the case will be more specific to study :)